### PR TITLE
Eslint: enforce valid JSdoc format

### PR DESCRIPTION
**Pull request by s-rigaud**
**Description**

Ensure that functions using JSdoc are properlly documented (no more parameter naming mistake and better editor linting support)

### CODE REVIEW

File: src/nodes/pmrem/PMREMNode.js [hunk 1]
```diff
@@ -203,6 +203,7 @@ class PMREMNode extends TempNode {
 	/**
 	 * The node's texture value.
 	 *
+	 * @param {Texture} value
```
Comment (Mugen87) [inline src/nodes/pmrem/PMREMNode.js:206]: I don't think this bit is right. We do not document a setter here but just a property. Properties don't use `param`.

File: src/nodes/core/Node.js [hunk 2]
```diff
@@ -531,9 +532,9 @@ class Node extends EventDispatcher {
 	/**
 	 * The method can be implemented to update the node's internal state before it is used to render an object.
 	 * The {@link Node#updateBeforeType} property defines how often the update is executed.
+	 * This method should return an optional bool that indicates whether the implementation actually performed an update or not (e.g. due to caching).
 	 *
 	 * @param {NodeFrame} frame - A reference to the current node frame.
-	 * @return {Boolean?} An optional bool that indicates whether the implementation actually performed an update or not (e.g. due to caching).
```
Comment (Mugen87) [inline src/nodes/core/Node.js:536]: Even though the node does not return a bool, `@return` should define the type and the description for this abstract method.

File: src/nodes/pmrem/PMREMNode.js (hunk 1 above)
Comment (s-rigaud) [reply src/nodes/pmrem/PMREMNode.js:206]: Well I was thinking the same at first but without this line, eslint hrow this `Missing JSDoc for parameter 'value'.eslint[valid-jsdoc]` 🤔

File: src/nodes/core/Node.js (hunk 2 above)
Comment (s-rigaud) [reply src/nodes/core/Node.js:536]: This also ends up throwing an eslint error as there is no direct return in this abstract function, that's why I move the annotation to a standard description sentence.

Comment (s-rigaud) [reply src/nodes/core/Node.js:536]: Settings the @abstract flag seems to fix the issue

File: src/nodes/pmrem/PMREMNode.js (hunk 1 above)
Comment (s-rigaud) [reply src/nodes/pmrem/PMREMNode.js:206]: Seems like some recommendations are to not document the setter at all, what do you think ? https://github.com/jsdoc/jsdoc/issues/973

Comment (Mugen87) [reply src/nodes/pmrem/PMREMNode.js:206]: The idea is to document setter/getter only once per name.

In this case I would remove the ESLint rule.

Comment (s-rigaud) [reply src/nodes/pmrem/PMREMNode.js]: Yes, the documentation is also available on the setter 
image

**Pull request merged**